sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/EventHubReactorAmqpConnection.java

```diff
@@ -101,19 +102,20 @@ public Mono<EventHubManagementNode> getManagementNode() {
      * @param linkName The name of the link.
      * @param entityPath The remote address to connect to for the message broker.
      * @param retryOptions Options to use when creating the link.
+     * @param clientIdentifier The identifier of client.
      * @return A new or existing send link that is connected to the given {@code entityPath}.
      */
     @Override
-    public Mono<AmqpSendLink> createSendLink(String linkName, String entityPath, AmqpRetryOptions retryOptions) {
-        return createSession(entityPath).flatMap(session -> {
-            logger.atVerbose()
-                .addKeyValue(ENTITY_PATH_KEY, entityPath)
-                .log("Get or create producer.");
-            final AmqpRetryPolicy retryPolicy = RetryUtil.getRetryPolicy(retryOptions);
-
-            return session.createProducer(linkName, entityPath, retryOptions.getTryTimeout(), retryPolicy)
-                .cast(AmqpSendLink.class);
-        });
+    public Mono<AmqpSendLink> createSendLink(String linkName, String entityPath, AmqpRetryOptions retryOptions, String clientIdentifier) {
```

Comment:
Do we need to update the public method for this? I was thinking since the client identifier is unique per EventHubClient, that it would be set in the connection rather than for a method parameter.

Reply:
If user set shareConnection true, and build several client with diferent identifier (or don't set identifier, default behavior is generate a unique identifier for each client), we set identifier in the connection, all links created by this connection will use the same identifier. I think identifier is for client, not for connection. Not sure set in connection is OK. And sorry for my ignorance of CSharp, I didn't find shareConnection in dotnet code.

Reply:
We can check in .NET SDK where it saves the identifier. Besides, we need to verify if changing this public method will it bring any impact.

Reply:
.NET SDK seems like only save identifier in client https://github.com/Azure/azure-sdk-for-net/blob/7a49112d5c364820b8b42c83d360ea1edf5eb7b3/sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs#L344-L352

use identifier as a parameter when create consumer https://github.com/Azure/azure-sdk-for-net/blob/7a49112d5c364820b8b42c83d360ea1edf5eb7b3/sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs#L487-L492